app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -163,12 +202,44 @@ def parse_deployment_name(name)
         name.sub(/^.*deployment=/, '')
       end
 
-      def parse_middleware_server(eap)
+      def parse_server_group_name(name)
+        name.sub(/^Domain Server Group \[/, '').chomp(']')
+      end
+
+      def parse_domain_server_name(name)
+        name.sub(/^.*\/server=/, '')
+      end
+
+      def parse_middleware_domain(domain)
```

Comment:
I think that several parse_X entity methods here can be shortened and have code reused by adding a parse_base method that parses properties that all entities have. something similar to [this](https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb#L703). wdyt?